Automattic/wp-calypso
# Adds Blaze dashboard standalone app

Related to SELFSERVE-412

## Proposed Changes

We are making the Advertising dashboard compatible to be used from a Standalone app, that will be consumed inside Jetpack (just like the stats page).

This PR adds a first version of the standalone app (based on the work done in the stats page), and also applies changes to make the advertising page compatible with Jetpack.

### Technical details of the changes:
**DSP Fetches to the API**
I changed to use the library `calypso/lib/wp` instead of directly calling the `request` method of the `wpcom-proxy-request`. It is the same on the inside because the wp library uses the `wpcom-proxy-request` handler to make the calls.

**Dynamic Advertising section URL prefix**
We need a dynamic URL prefix for the Advertising section because we won't use that prefix inside Jetpack (and on Woo in the future). I added a utility method that generates the final URL by getting the URL prefix from a configuration property.
I only spread the changes inside our Advertising module; there are links outside this module that still uses the `/advertising` hardcode prefix, but those aren't a problem because they won't line inside Jetpack or Woo.

**Refactor of the controller.js in promote-post-i2 section**
Because of how Webpack works in the Blaze Dashboard app, we need to be careful with the imports in the promote post section. 
I moved the `redirectToPrimarySite` function from the `controller.js` to the `index.js` because one of the components used in that function forced Webpack to bundle components that we don't need, and some of them conflict with the app breaking styles.
For now, I only detected that importing `calypso/controller` will make our final bundle bigger, forcing us to import the Layout component that will conflict with our custom Jetpack Layout. Let's try having all the logic that needs the `calypso/controller` living inside the index.js instead of the controller.

**Blaze Dashboard new App**
I am adding a new application inside the `/apps` folder. This app is based on the `odyssey-stats` app (an app used inside Jetpack to render the stats page from Calypso). 
have this in mind while reviewing the app, it has some changes from the other, and is still not finished.


## Testing Instructions
* Open the Live preview
* Login as A8C user
* Check out Tools > Advertising
* Verify that the page loads the products/campaigns correctly
* Click on the Promote button on one of the posts
* Verify that the DSP create campaign widget loads correctly
* Click on Cancel and then confirm the modal
* Verify that you get redirected to the post page.

## Pre-merge Checklist
